sale_require_purchase_order_number/models/stock_picking.py

```diff
@@ -9,42 +9,42 @@
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
-    @api.one
     @api.depends('group_id', 'sale_id')
-    def _get_purchase_order_number(self):
-        if self.manual_purchase_order_number:
-            self.purchase_order_number = self.manual_purchase_order_number
-        else:
-            self.purchase_order_number = self.sale_id.purchase_order_number
+    def _compute_purchase_order_number(self):
```

Comment:
Not requrired really: just for standard. move methods definitions after the fields definitions